core/state_transition.go

```diff
@@ -259,6 +259,10 @@ func (st *StateTransition) preCheck() error {
 		st.gas += st.msg.Gas() // Add gas here in order to be able to execute calls.
 		// Don't touch the gas pool for system transactions
 		if st.msg.IsSystemTx() {
+			if st.evm.ChainConfig().IsOptimismRegolith(st.evm.Context.Time) {
+				return fmt.Errorf("%w: address %v", ErrSystemTxNotSupported,
+					st.msg.From().Hex())
```

Comment:
```suggestion
			if st.evm.ChainConfig().IsOptimismRegolith(st.evm.Context.Time) {
				return fmt.Errorf("%w: address %s", ErrSystemTxNotSupported,
					st.msg.From())
```
`common.Address` implements a stringer which will format it nicely

Reply:
I'm happy to change this, but the pattern for other errors includes the `.Hex()` call. Not sure if that's deliberately so the full hash is always available (`common.Hash.TerminalString()` would truncate it which can be unhelpful when debugging) or if this code predates having a nice String implementation...

Reply:
So `Hex()` encodes with the lower/upper-case checksum thing, and `String` just calls `Hex`: https://github.com/ethereum/go-ethereum/blob/a609e7b81fbf360745b06a35227da52cc50c2d1b/common/types.go#L244
I guess `Address.String()` didn't always do the checksum thing, or `String` wasn't implemented on `Address`.

So it's not necessary to do `Hex()`, but you're right about the consistency with other calls, so NVM changing it.